### DISCUSSION TITLE
Suggestion: use Radix UI's <Slot /> instead of the ButtonLink component

### DISCUSSION THREAD

**Discussion by andrecasal** [epicweb-dev/epic-stack#195] [Ideas]
We could use the [Slot](https://www.radix-ui.com/docs/primitives/utilities/slot) pattern to

```tsx
// your-button.jsx
import React from 'react';
import { Slot } from '@radix-ui/react-slot';

function Button({ asChild, ...props }) {
  const Comp = asChild ? Slot : 'button';
  return <Comp {...props} />;
}
```

And instead of having a `ButtonLink` and `LabelButton` components, we could do:
```tsx
import { Button } from './your-button';

export default () => (
  <>
    <Button asChild>
      <a href="/contact">Contact</a>
    </Button>

    <Button asChild>
      <label>Contact</a>
    </Button>
  </>
);
```
We could then do away with the `getButtonClassName()`and colocate everything in the `Button` component.

**Comment by kentcdodds** [comment]
I think this is a fair idea. However I believe shadcn has a button component and I've not yet looked at how that one is built. Let's solidify that first.

**Comment by kentcdodds** [comment]
This is done now
